src/analyzer/analyzerwaveform.cpp

```diff
@@ -353,6 +351,15 @@ void AnalyzerWaveform::storeResults(TrackPointer tio) {
             m_waveform,
             m_waveformSummary);
 
+    // Set waveforms on track AFTER they'been written to disk in order to have
+    // a consistency when OverviewCache asks AnalysisDAO for a waveform summary.
+    if (m_waveform) {
+        tio->setWaveform(m_waveform);
+    }
+    if (m_waveformSummary) {
```

Comment:
Same here, write the waveform uncoditionally?

Reply:
okay. I didn't consider that the analyzer may a produce null waveforms.
done.